### ISSUE TITLE
Wiki page describing Part-o-Magic

### ISSUE THREAD

**Issue by luzpaz** [DeepSOIC/Part-o-magic#25]
Hey @DeepSOIC 
Just a heads up that I added your Part-o-Magic icon to https://www.freecadweb.org/wiki/Template:DevWorkbenches 

Yorik was saying that he'd like every single Addon to have a wiki page. I know you haven't gotten around to documenting this workbench... it would be really awesome if you did. Do you need help ?

**Comment by DeepSOIC** [comment]
> Do you need help ?

That'd be lovely! I have trouble finding time to work on my workbenches now, be it code or documentation.

**Comment by jgriessen** [comment]
I'd like to try to learn to use part-o-magic.  After reading old threads it seems you hit an impasse in April 2017 and could not get it merged into freecad, so slowed down on App.ActiveContainer.  What should I do right now?  Ask for help from others and use part design in FreeCAD 0.18, Libs: 0.18R14726 (Git)?
Or is part-o-magic usable and some people here will help me learn it without your input since you are busy?

**Comment by DeepSOIC** [comment]
> What should I do right now?
You decide!

Ask your questions, either here in PoM issue tracker, or in FreeCAD forum. On FC forum, ping me with a quote, so that I notice. I'll answer as many as I can, no problem.

Your questions may push me over that write-the-docs barrier!

**Comment by luzpaz** [comment]
> Your questions may push me over that write-the-docs barrier!  

@jgriessen this is your chance!
![image](https://user-images.githubusercontent.com/4140247/46770003-b5738e80-ccbb-11e8-9d32-84be4a94e7d9.png)

**Comment by jgriessen** [comment]
I'm good at writing also.  I'll help.

For freecad forum I should ping you as forum reply with quote only, or also do a forum personal message?

John Griessen

Parting red sea!  Funny!  :-)

**Comment by jgriessen** [comment]
Back in Feb 19, 2017, you said, "You can still make a sketch on a face of a Cube, be it in Module or not. You just need to make a new sketch from a Sketcher workbench, not from PartDesign workbench."
Is this still a rule?  Are there other "Don't do this or fails." rules to follow while building things with part-o-magic?

**Comment by DeepSOIC** [comment]
> Is this still a rule? 

Yes. It's not a part-o-magic rule, it's a general "if you want a sketch for anything but partdesign" rule.

> Are there other "Don't do this or fails." rules to follow while building things with part-o-magic?
Yes. I don't quite have a list. So let me try.

**Comment by DeepSOIC** [comment]
* don't make cross-container links. 

That is, when you want to do something, make sure the correct container is active.

Often, you can (and should) use whole body/module/shapegroup for an operation.

* if you need cross-container links, use shapebinder or ghost

** replacement tool can assist with inserting shapebinders/ghosts into cross-container links.

* start your project with a container

it's much harder to pack up a container after the fact. I hope to introduce some tools to help with that, eventually.

* when duplicating/copy-pasting, avoid copying dependencies. Tends to create a big mess.

* (this is a bug) switch off all x-rays before saving a project. Otherwise, the objects are saved in semi-x-rayed state, and have to be recovered manually by switching display mode and zeroing out transparency value.

**Comment by jgriessen** [comment]
Thanks,  I will write this list into my howto_summary.txt and publish it on the https://www.freecadweb.org/wiki/Template:DevWorkbenches  wiki page as I test it out.  It's been 28 days, but now I am going to be spending time on this for a few days and get some results.

**Comment by jgriessen** [comment]
"Often, you can (and should) use whole body/module/shapegroup for an operation."  Not understanding well.  Is this an example:  I am wanting a boolean subtract.  I make a body within the main container, enclosure_back, active and make a rectangular shape, (that I want to subtract).  I make enclosure_back active again, select the rectangular shape and additive pipe and create a boolean of them.

Alternatively: I could make top container active, make a rectangular shape, (that I want to subtract).   I make top active again, select the rectangular shape and enclosure_back, and create a boolean of them.

Are you saying use Alternatively as a rule?

**Comment by jgriessen** [comment]
"avoid copying dependencies. "  that seems tricky.  Not sure how to do.

**Comment by DeepSOIC** [comment]
> Often, you can (and should) use whole body/module/shapegroup for an operation." Not understanding well. Is this an example: I am wanting a boolean subtract. I make a body within the main container, enclosure_back, active and make a rectangular shape, (that I want to subtract). I make enclosure_back active again, select the rectangular shape and additive pipe and create a boolean of them.
Alternatively: I could make top container active, make a rectangular shape, (that I want to subtract). I make top active again, select the rectangular shape and enclosure_back, and create a boolean of them.

It's hard to say without having your project. I'm having a hard time understanding you. But it seems like both ways you describe are valid.

**Comment by DeepSOIC** [comment]
>           "avoid copying dependencies. "  that seems tricky.  Not sure how to do.

When you duplicate something, FC often asks "do you also want to duplicate dependencies?". If you use it to duplicate Modules, Parts, Bodies, especially those that depend on something external, chaos can result.

**Comment by jgriessen** [comment]
OK, I will try copying without dependencies and then put back the orientations etc.  I guess spreadsheet is a dependency that cannot copy, or select it and see?  I will try it and always answer"No." to FC asking "do you also want to duplicate dependencies?".

**Comment by jgriessen** [comment]
I'm learning how to select all the planes in Origin003 etcetera to get a copy to work without errors such as, "Lost link to 'X_Axis004' while loading, maybe an object was not loaded correctly"   Now to write this up and keep going creating reused chunks of what I drew before.  So far, so good.

**Comment by jgriessen** [comment]
I've tried adapting the example of https://forum.freecadweb.org/viewtopic.php?t=20698&start=50
subtractive shape container, to my design and don't get boolean subtract and do get errors.
I put the details on freecadweb:   https://forum.freecadweb.org/viewtopic.php?f=22&t=32559

**Comment by jgriessen** [comment]
I'm learning how to select all the planes in Origin003 etcetera to get a copy to work without errors such as, "Lost link to 'X_Axis004' while loading, maybe an object was not loaded correctly"   Now to write this up and keep going creating reused chunks of what I drew before.  So far, so good.

**Comment by jgriessen** [comment]
subtractive-shape-save-test3 has no offset position value  in its property list.  Have I created it the wrong way?
box that was there from your file is same way -- no offset position value  in its property list.  More details on https://forum.freecadweb.org/viewtopic.php?f=22&t=32559&start=20
